components/event-service/internal/externalapi/genericeventhandler_test.go

```diff
@@ -20,101 +20,116 @@ func TestNewEventsHandler(t *testing.T) {
+			wantErrorType:             shared.ErrorTypeValidationViolation,
+			wantErrorField:            shared.FieldData,
 		},
 		{
-			name:           "invalid event empty field data",
-			givenPayload:   `{"event-type":"order.created", "event-type-version":"v1", "event-time":"2018-11-02T22:08:41+00:00", "data":""}`,
-			wantCode:       http.StatusBadRequest,
-			wantErrorType:  shared.ErrorTypeValidationViolation,
-			wantErrorField: shared.FieldData,
+			name:                      "invalid event empty field data",
+			givenPayload:              `{"event-type":"order.created", "event-type-version":"v1", "event-time":"2018-11-02T22:08:41+00:00", "data":""}`,
+			withEventMeshResponseCode: http.StatusOK,
+			wantCode:                  http.StatusBadRequest,
+			wantErrorType:             shared.ErrorTypeValidationViolation,
+			wantErrorField:            shared.FieldData,
 		},
 		{
-			name:         "valid event without event-id",
-			givenPayload: `{"event-type":"order.created", "event-type-version":"v1", "event-time":"2018-11-02T22:08:41+00:00", "data":{ "order-number":123}}`,
-			wantCode:     http.StatusOK,
+			name:                      "error while sending a CE to event-mesh",
+			givenPayload:              `{"event-id":"8954ad1c-78ed-4c58-a639-68bd44031de0", "event-type":"order.created", "event-type-version":"v1", "event-time":"2018-11-02T22:08:41+00:00", "data":{ "order-number":123}}`,
+			withEventMeshResponseCode: http.StatusInternalServerError,
+			wantCode:                  http.StatusInternalServerError,
+			wantErrorType:             shared.ErrorTypeInternalServer,
 		},
 		{
-			name:         "valid event with event-id",
-			givenPayload: `{"event-id":"8954ad1c-78ed-4c58-a639-68bd44031de0", "event-type":"order.created", "event-type-version":"v1", "event-time":"2018-11-02T22:08:41+00:00", "data":{ "order-number":123}}`,
-			wantCode:     http.StatusOK,
-			wantEventID:  "8954ad1c-78ed-4c58-a639-68bd44031de0",
+			name:                      "valid event without event-id",
+			givenPayload:              `{"event-type":"order.created", "event-type-version":"v1", "event-time":"2018-11-02T22:08:41+00:00", "data":{ "order-number":123}}`,
+			withEventMeshResponseCode: http.StatusOK,
+			wantCode:                  http.StatusOK,
+		},
+		{
+			name:                      "valid event with event-id",
+			givenPayload:              `{"event-id":"8954ad1c-78ed-4c58-a639-68bd44031de0", "event-type":"order.created", "event-type-version":"v1", "event-time":"2018-11-02T22:08:41+00:00", "data":{ "order-number":123}}`,
+			withEventMeshResponseCode: http.StatusOK,
+			wantCode:                  http.StatusOK,
+			wantEventID:               "8954ad1c-78ed-4c58-a639-68bd44031de0",
 		},
 	}
 
+	var mux *http.ServeMux
+	var closeFn meshtesting.CloseFunction
 	for _, test := range tests {
+
 		t.Run(test.name, func(t *testing.T) {
+			mux, closeFn = setupEventMeshServer(source, v1Endpoint, requestSize, test.withEventMeshResponseCode, t)
```

Comment:
Can this run in parallel? Also applies for the test itself **TestNewEventsHandler**.

Reply:
I tried to execute it in parallel(`t.Parallel()`) which works intermittently. Executing 100 runs makes the code run into panics. 

Parallelizing the tests is a great idea. Since the tests are indeterministic if we execute them in parallel, I think it is better if we take care of this in a separate issue and tackle the problem holistically for the entire test-suite.

Reply:
The panic was due to the sharing of variables that we fixed and executed the tests again. To our surprise, the gain was not evident.
```
➜ go test -count 1000 github.com/kyma-project/kyma/components/event-service/internal/externalapi (not in parallel)
ok      github.com/kyma-project/kyma/components/event-service/internal/externalapi      10.071s
~ at ☸️  gke-release-1-11-1-rc1.a.build.kyma-project.io took 13s
➜ go test -count 1000 github.com/kyma-project/kyma/components/event-service/internal/externalapi (in parallel)
ok      github.com/kyma-project/kyma/components/event-service/internal/externalapi      34.754s
```

Hence we decided to stay away from parallel for now.